### ISSUE TITLE
[Bug]: setting patchA of Annotation.arrow_patch has no effect

### ISSUE THREAD

**Issue by Phlya** [matplotlib/matplotlib#28316]
### Bug summary

I am trying to improve some edge cases for plotting arrows in adjustText. Without going into details, sometimes I have to use ax.annotate to add arrows instead of directly using FancyArrowPatch. However in this case the arrows are not associated with the text objects and strike through them. I thought I could modify the arrow patch of the Annotation object to point to the corresponding Text object. The object appears to be set correctly, however this doesn't fix the clipping of the arrow! With a regular FancyArrowPatch this procedure works just as expected. Interestingly, if I update other properties (color) it works perfectly.

(As an aside, why do the two arrows have a different style in the example below?)

### Code for reproduction

```Python
import matplotlib.pyplot as plt
from matplotlib import patches
xytext = 0.2, 0.5
f, ax = plt.subplots()
ann = ax.annotate('', xy=(0.5, 0.6), xytext=xytext, arrowprops=dict(arrowstyle="->"))
ann_arrowpatch = ann.arrow_patch
print(ann_arrowpatch.patchA)
text = plt.text(xytext[0], xytext[1], 'Text', ha='center', va='center')
ann_arrowpatch.set_patchA(text)
print(ann_arrowpatch.patchA)

ann_arrowpatch.set_color('b')

arrowpatch = patches.FancyArrowPatch(posA=xytext, posB=(0.5, 0.4), arrowstyle="->", color='r')
ax.add_patch(arrowpatch)
print(arrowpatch.patchA)
arrowpatch.set_patchA(text)
print(arrowpatch.patchA)
```

### Actual outcome

None
Text(0.2, 0.5, 'Text')
None
Text(0.2, 0.5, 'Text')

![image](https://github.com/matplotlib/matplotlib/assets/2895034/4a38fbd0-a304-4201-9e36-12799215df5f)

### Expected outcome

Expect that the blue arrow is also clipped by the Text

### Operating system

OS/X

### Matplotlib Version

3.8.0

### Matplotlib Backend

both inline and ipympl

### Python version

3.11.7

### Jupyter version

6.4.5

### Installation

conda

**Comment by Phlya** [comment]
Relevant issue in adjustText https://github.com/Phlya/adjustText/issues/175

**Comment by timhoffm** [comment]
This is indeed surprising. It's even better to see when using `]->` arrows. Also arrow scales are significantly different in the two approaches. There's something very fishy here.

![grafik](https://github.com/matplotlib/matplotlib/assets/2836374/f7363aae-2cb2-4ac1-8369-f91d19ddaec7)

**Comment by dikshajangra12918-oss** [comment]
It seems like the arrow position is correctly computed at creation time but not updated when the text or its extent changes later.
Would it make sense to trigger a recomputation of the arrow position after seting patchA, based on the current text extent?
Or is there a design reason why annotation does not update in the same way as FancyArrowPatch here?

**Comment by timhoffm** [comment]
You are welcome to investigate. I project, nobody will be able to answer that question off the top of their head.

**Comment by timhoffm** [comment]
@Phlya why are you doing `ax.annotate('', xy=(0.5, 0.6), xytext=xytext, arrowprops=dict(arrowstyle="->"))` and then trying to patch in another text?

This does not work as `Annotation` provides a text itself and "owns" the patchA of the associated arrow, i.e. it sets it at draw time to the bounding box of it's own text, which is empty in your case.

**Comment by Phlya** [comment]
It does seem illogical, but I think the issue is that I couldn't find a way to get the bbox of the text patch and/or a way to modify it... It's been a few years since I wrote that though.

**Comment by Phlya** [comment]
Ah, semi-scratch that (the above is also an issue, which prevents adjustText from using Annotation objects as input instead of Text).

Looking into it a bit now, I think actually the idea is that I wanted to actually use the original provided Text objects and maintain all of their properties (fonts, colours, etc) since they are provided by the user, and not look for a way to replicate all of that when creating Annotation objects instead of the provided Texts.

**Comment by Phlya** [comment]
FYI here is the code: https://github.com/Phlya/adjustText/blob/aab1e1915f0b995b0b0e53811f6aeedfabbe618a/adjustText/__init__.py#L781

**Comment by timhoffm** [comment]
Thanks for the clarification. So actually you want an arrow from "custom Artist" (your text object) to some point. That's what `ConnectionPatch` is for. Unfortunately it's currently less flexible than the transforms in Annotation, which makes it not yet a drop-in replacement, but the plan is to enhance that https://github.com/matplotlib/matplotlib/issues/26440.

**Comment by dikshajangra12918-oss** [comment]
Thanks for the clarification sir! I understand now that ConnectionPatch is the right long-term solution. I will look into issue #26440. Should I close this PR?

**Comment by timhoffm** [comment]
@dikshajangra12918-oss yes, please the related PR it's not the correct fix fox this scenario.
